Library/Formula/infer.rb

```diff
@@ -0,0 +1,60 @@
+class Infer < Formula
+  desc "Facebook Infer is a static analysis tool - if you give Infer some Objective-C, Java, or C code, it produces a list of potential bugs."
+  homepage "https://fbinfer.com/"
+  url "https://github.com/facebook/infer/releases/download/v0.1.0/infer-osx-v0.1.0.tar.xz"
+  sha256 "7a611989ceb4ece0882b8f62ed574fa0e59b48425cede20144b854a0918e1757"
+
+  def install
+    bin.install_symlink prefix/"infer"
+    libexec.install Dir["*"]
+    bin.install_symlink Dir["#{libexec}/infer/infer/bin/infer"]
+    inreplace "#{libexec}/infer/infer/bin/infer", 'os.path.dirname(__file__)', 'os.path.dirname(os.path.realpath(__file__))'
+  end
+
+  test do
+    # Java test
+    expected = <<eos
+Starting analysis (Infer version v0.1.0)
```

Comment:
You can indent this at the same level as the parent for readability and use the `undent` method on the string:

```ruby
  expected = <<-EOS.undent
  ...
  EOS
```